src/Microsoft.QualityGuidelines.Analyzers/Core/MarkMembersAsStatic.Fixer.cs

```diff
@@ -22,9 +29,33 @@ public sealed override FixAllProvider GetFixAllProvider()
 
         public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
         {
-            // This is to get rid of warning CS1998, please remove when implementing this analyzer
-            await Task.Run(() => { }).ConfigureAwait(false);
-            throw new NotImplementedException();
+            SyntaxNode root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            SyntaxNode node = root.FindNode(context.Span);
+            var model = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+            var symbol = model.GetDeclaredSymbol(node, context.CancellationToken);
+
+            context.RegisterCodeFix(new MarkMembersAsStaticAction(
+                    string.Format(MicrosoftQualityGuidelinesAnalyzersResources.MarkMembersAsStaticCodeFix, symbol.Name),
+                    async ct => await RemoveStaticModifier(context.Document, root, node, ct).ConfigureAwait(false)
+                ), context.Diagnostics.Single());
+        }
+
+        private Task<Document> RemoveStaticModifier(Document document, SyntaxNode root, SyntaxNode node, CancellationToken ct)
+        {
+            ct.ThrowIfCancellationRequested();
+            var syntaxGenerator = SyntaxGenerator.GetGenerator(document);
+            var madeStatic = syntaxGenerator.WithModifiers(node, DeclarationModifiers.Static);
```

Comment:
One of the side effects of this would be that all the references to instance members would now cause compiler errors. 

```
class C
{
    static int i;
    public int M1()
    {
        return i;
    }

    int M2(C d)
    {
        return d.M1();
    }

    int M3(C c)
    {
        return GetDerived(c).M1();
    }

    C GetDerived(C c)
    {
        System.Console.Write("GetDerived executed");
        return c;
    }
}
```

Switching M1 to be a static method fires CS0176 inside M2 and M3.
Ideally we would fix the reference sites to use type name as the receiver instead of the instance access, but this could cause side effects, such as M3 would not execute GetDerived() anymore.

It is tricky how to implement such a code fix without causing any side effects. Essentially we will need to offer a code fix for CS0176